executor/join_test.go

```diff
@@ -387,6 +387,14 @@ func (s *testSuite) TestSubquery(c *C) {
 	result.Check(testkit.Rows("1 1", "2 2", "3 3"))
 
 	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(id int, v int)")
+	tk.MustExec("insert into t values(1, 1), (2, 2), (3, 3)")
+	result = tk.MustQuery("select * from t where v=(select min(t1.v) from t t1, t t2, t t3 where t1.id=t2.id and t2.id=t3.id and t1.id=t.id)")
+	result.Check(testkit.Rows("1 1", "2 2", "3 3"))
+	result = tk.MustQuery("select exists (select t.id from t where s.id < 2 and t.id = s.id) from t s")
```

Comment:
For this case, I think the problem is that the subquery condition `s.id < 2` should not be pushed to outer table.
For all filter condition on outer table, it doesn't need to return matched.